iati_datastore/iatilib/crawler.py

```diff
@@ -166,30 +113,17 @@ def fetch_resource(resource):
     :param resource:
     :return:
     '''
-    headers = {
-        'User-Agent': 'codeforIATI datastore classic',
-    }
-    if resource.last_succ:
-        headers['If-Modified-Since'] = http_date(resource.last_succ)
-    if resource.etag:
-        headers["If-None-Match"] = resource.etag
-    resp = requests.get(resource.url, headers=headers)
-    resource.last_status_code = resp.status_code
-    resource.last_fetch = datetime.datetime.utcnow()
-    if resp.status_code == 200:
-        if isinstance(resp.content, bytes):
-            resource.document = resp.content
-        else:
-            resource.document = resp.content.encode()
-        if "etag" in resp.headers:
-            resource.etag = resp.headers.get('etag')
-        else:
-            resource.etag = None
-        resource.last_succ = datetime.datetime.utcnow()
-        resource.last_parsed = None
-        resource.last_parse_error = None
-    if resp.status_code == 304:
-        resource.last_succ = datetime.datetime.utcnow()
+    dataset = Dataset.query.get(resource.dataset_id)
+    fname = '__iatikitcache__/registry/data/{0}/{1}.xml'.format(
+        dataset.publisher, dataset.name)
+    with open(fname, 'rb') as f:
+        content = f.read()
+
+    resource.document = content
+    resource.last_succ = iatikit.data().last_updated
```

Comment:
It looks like `last_fetch` does not get set anywhere? (as opposed to `last_succ`) -- also `last_status_code` is not set.

Does `iatikit` store data about failed downloads?

Reply:
I think this is the reason for the following results in these endpoints?

`/api/1/about/`
![image](https://user-images.githubusercontent.com/688113/102672312-b4d6bc80-4190-11eb-8c5e-a1e72938de50.png)

`/api/1/about/fcdo-bd/`
![image](https://user-images.githubusercontent.com/688113/102672353-d33cb800-4190-11eb-9de5-e5358c46dfe5.png)

Reply:
Oooh – good spot…